plugins/datadog-standalone/datadog-standalone.go

```diff
@@ -0,0 +1,218 @@
+	}
+)
+
+// DataPoint is a tuple of [UNIX timestamp, value]. This has to use floats
+// because the value could be non-integer.
+type DDDataPoint [2]float64
+
+// Metric represents a collection of data points that we might send or receive
+// on one single metric line.
+type DDMetric struct {
+	Metric string        `json:"metric,omitempty"`
+	Points []DDDataPoint `json:"points,omitempty"`
+	Type   *string       `json:"type,omitempty"`
+	Host   *string       `json:"host,omitempty"`
+	Tags   []string      `json:"tags,omitempty"`
+	Unit   string        `json:"unit,omitempty"`
+}
+
+// Series represents a collection of data points we get when we query for timeseries data
+type DDSeries struct {
+	Series []DDMetric `json:"series"`
+}
+
+// A successful metrics API result
+type DDResult struct {
+	Status   string   `json:"status,omitempty"`
+	Errors   []string `json:"errors,omitempty"`
+	Warnings []string `json:"warnings,omitempty"`
+}
+
+func main() {
+	log.Info("Starting Standalone DataDog DC/OS metrics plugin")
+	datadogPlugin, err := plugin.New(
+		plugin.PluginName("standalone-datadog"),
+		plugin.ExtraFlags(pluginFlags),
+		plugin.ConnectorFunc(datadogConnector))
+
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	log.Fatal(datadogPlugin.StartPlugin())
+}
+
+func messagesToSeries(messages []producers.MetricsMessage) *DDSeries {
+
+	series := new(DDSeries)
+
+	for _, message := range messages {
+		dimensions := &message.Dimensions
+		host := &dimensions.Hostname
+		messageTags := []string{}
+		addMessageTag := func(key, value string) {
+			if len(value) > 0 {
+				messageTags = append(messageTags, fmt.Sprintf("%s:%s", key, value))
+			}
+		}
+
+		for name, value := range dimensions.Labels {
+			addMessageTag(name, value)
+		}
+		addMessageTag("mesosId", dimensions.MesosID)
+		addMessageTag("clusterId", dimensions.ClusterID)
+		addMessageTag("containerId", dimensions.ContainerID)
+		addMessageTag("executorId", dimensions.ExecutorID)
+		addMessageTag("frameworkName", dimensions.FrameworkName)
+		addMessageTag("frameworkId", dimensions.FrameworkID)
+		addMessageTag("frameworkRole", dimensions.FrameworkRole)
+		addMessageTag("frameworkPrincipal", dimensions.FrameworkPrincipal)
+		addMessageTag("hostname", dimensions.Hostname)
+
+		for _, datapoint := range message.Datapoints {
+			t, err := parseTimestamp(datapoint.Timestamp)
+			if err != nil {
+				log.Errorf("Could not parse timestamp '%s': %v", datapoint.Timestamp, err)
+				continue
+			}
+			v, err := toFloat64(datapoint.Value)
+			if err != nil {
+				log.Error(err)
+				continue
+			}
+
+			datapointTags := []string{}
+			addDatapointTag := func(key, value string) {
+				datapointTags = append(datapointTags, fmt.Sprintf("%s:%s", key, value))
+			}
+			for name, value := range datapoint.Tags {
+				addDatapointTag(name, value)
+			}
+
+			metric := DDMetric{
+				Metric: datapoint.Name,
+				Points: []DDDataPoint{{float64(t.Unix()), v}},
+				Tags:   append(messageTags, datapointTags...),
```

Comment:
We found out the other week that we had some slight naming issues with tag names when we were looking at the Librato results. I believe we had a scenario where we could end up with two tags `containerId` from the messages and then `container.id` from the datapoint tags.  Perhaps we should figure that out as a part of this changeset?